pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -3,29 +3,35 @@ package dispatcher
 import (
 	"context"
 	"fmt"
+	"sync"
+
 	"github.com/Shopify/sarama"
 	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+	commonconfig "knative.dev/eventing-kafka/pkg/common/config"
 	"knative.dev/eventing-kafka/pkg/common/kafka/consumer"
+	"knative.dev/eventing-kafka/pkg/common/kafka/util"
 	"knative.dev/eventing-kafka/pkg/common/metrics"
 	eventingduck "knative.dev/eventing/pkg/apis/duck/v1"
 	"knative.dev/eventing/pkg/channel"
-	"sync"
 )
 
 // Define A Dispatcher Config Struct To Hold Configuration
 type DispatcherConfig struct {
-	Logger               *zap.Logger
-	ClientId             string
-	Brokers              []string
-	Topic                string
-	Username             string
-	Password             string
-	ChannelKey           string
-	StatsReporter        metrics.StatsReporter
-	ExponentialBackoff   bool
-	InitialRetryInterval int64
-	MaxRetryTime         int64
+	Logger                 *zap.Logger
+	ClientId               string
+	Brokers                []string
+	Topic                  string
+	Username               string
+	Password               string
+	ChannelKey             string
+	StatsReporter          metrics.StatsReporter
+	ExponentialBackoff     bool
+	InitialRetryInterval   int64
+	MaxRetryTime           int64
+	SaramaConfig           *sarama.Config
+	currentSubscriberSpecs []eventingduck.SubscriberSpec
 }
```

Comment:
Would call this `SubscriberSpecs` (caps) for consistency and there's no alternative to "current"